contracts/SimpleManager.sol

```diff
@@ -154,48 +171,15 @@ contract SimpleManager is OwnableUpgradeable {
             token1Decimals = IDecimals(token1).decimals();
         }
 
-        if (mintsLength > 0) {
-            checked = new uint24[](mintsLength);
-            oraclePrice = vaultInfo.oracle.getPrice0();
-        }
-
-        for (uint256 i; i < mintsLength; ++i) {
-            if (
-                _includes(
-                    rebalanceParams_.mints[i].range.feeTier,
-                    checked,
-                    increment
-                )
-            ) continue;
-
-            IUniswapV3Pool pool = IUniswapV3Pool(
-                _getPool(
-                    token0,
-                    token1,
-                    rebalanceParams_.mints[i].range.feeTier
-                )
-            );
-
-            uint256 sqrtPriceX96;
-
-            (sqrtPriceX96, , , , , , ) = pool.slot0();
-
-            uint256 poolPrice = FullMath.mulDiv(
-                sqrtPriceX96 * sqrtPriceX96,
-                10 ** token0Decimals,
-                2 ** 192
-            );
-
-            _checkDeviation(
-                poolPrice,
-                oraclePrice,
-                vaultInfo.maxDeviation,
-                token1Decimals
-            );
-
-            checked[increment] = rebalanceParams_.mints[i].range.feeTier;
-            increment++;
-        }
+        _checkPoolsPrices(
```

Comment:
Do we still need to conduct a price check prior to the rebalance, considering there is already a post-rebalance check? Seems necessary, the post-rebalance check should be enough.